packages/next-auth/src/core/index.ts

```diff
@@ -94,13 +94,21 @@ export async function NextAuthHandler<
     assertionResult.forEach(logger.warn)
   } else if (assertionResult instanceof Error) {
     // Bail out early if there's an error in the user config
-    const { pages, theme } = userOptions
     logger.error(assertionResult.code, assertionResult)
 
+    const htmlPages = ["signin", "verify-request", "signout"]
+    if (!htmlPages.includes(req.action) || req.method !== "GET") {
```

Comment:
Should we check for header like `Accept: application/json` instead? 🤔 That way we'll be sure that we don't miss any requests in the check

Reply:
Some requests will set something like `Accept: */*` so we cannot rely on that, unfortunately.